src/System.Security.Cryptography.Algorithms/tests/RandomNumberGeneratorTests.netcoreapp.cs

```diff
@@ -92,5 +94,132 @@ public static void Fill_RandomDistribution()
 
             RandomDataGenerator.VerifyRandomDistribution(random);
         }
+
+        [Theory]
+        [InlineData(10, 10)]
+        [InlineData(10, 9)]
+        [InlineData(-10, -10)]
+        [InlineData(-10, -11)]
+        public static void GetInt32_LowerAndUpper_InvalidRange(int fromInclusive, int toExclusive)
+        {
+            Assert.Throws<ArgumentException>(() => RandomNumberGenerator.GetInt32(fromInclusive, toExclusive));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-10)]
+        public static void GetInt32_Upper_InvalidRange(int toExclusive)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => RandomNumberGenerator.GetInt32(toExclusive));
+        }
+
+        [Theory]
+        [InlineData(1 << 1)]
+        [InlineData(1 << 4)]
+        [InlineData(1 << 16)]
+        [InlineData(1 << 24)]
+        public static void GetInt32_PowersOfTwo(int toExclusive)
+        {
+            for (int i = 0; i < 10; i++)
+            {
+                int result = RandomNumberGenerator.GetInt32(toExclusive);
+                Assert.InRange(result, 0, toExclusive - 1);
+            }
+        }
+
+        [Theory]
+        [InlineData((1 << 1) + 1)]
+        [InlineData((1 << 4) + 1)]
+        [InlineData((1 << 16) + 1)]
+        [InlineData((1 << 24) + 1)]
+        public static void GetInt32_PowersOfTwoPlusOne(int toExclusive)
+        {
+            for (int i = 0; i < 10; i++)
+            {
+                int result = RandomNumberGenerator.GetInt32(toExclusive);
+                Assert.InRange(result, 0, toExclusive - 1);
+            }
+        }
+
+        [Fact]
+        public static void GetInt32_FullRange()
+        {
+            int result = RandomNumberGenerator.GetInt32(int.MinValue, int.MaxValue);
+            Assert.NotEqual(0, result);
```

Comment:
There's a 1 in 4 billion chance this fails.  If you want to assert something you should just assert it wasn't int.MaxValue